src/app/record/components/research-resources/research-resources.component.ts

```diff
@@ -0,0 +1,208 @@
+          putCode
+        )
+        .pipe(first())
+        .subscribe(
+          (data) => {
+            this.detailsResearchResources.push({
+              putCode: putCode,
+              researchResource: data,
+            })
+            researchResource.showDetails = true
+          },
+          (error) => {
+            console.log('getDetailsError', error)
+          }
+        )
+    } else {
+      this._recordResearchResourceService
+        .getResearchResourceById(putCode)
+        .pipe(first())
+        .subscribe(
+          (data) => {
+            const research: ResearchResource = data
+            this.detailsResearchResources.push({
+              putCode: putCode,
+              researchResource: research,
+            })
+            researchResource.showDetails = true
+            research.hosts.forEach((host) => {
+              this.getOrganizationDisambiguatedDetails(
+                null,
+                host.disambiguationSource,
+                host.orgDisambiguatedId
+              )
+            })
+          },
+          (error) => {
+            console.log('getDetailsError', error)
+          }
+        )
+    }
+  }
+
+  getOrganizationDisambiguatedDetails(
+    item: Item,
+    disambiguationSource,
+    orgDisambiguatedId
+  ): void {
+    this._organizationsService
+      .getOrgDisambiguated(disambiguationSource, orgDisambiguatedId)
+      .pipe(first())
+      .subscribe((organizationDisambiguated) => {
+        this.detailsOrgDisambiguated.push({
+          disambiguationSource,
+          orgDisambiguatedId,
+          orgDisambiguated: organizationDisambiguated,
+        })
+        if (item) {
+          item.showDetails = true
+        }
+      })
+  }
+
+  getResearchResource(putCode: number): ResearchResource {
+    if (this.detailsResearchResources.length === 0) {
+      return null
+    }
+
+    return this.detailsResearchResources
+      .filter((value) => value.putCode === putCode)
+      .map((value) => {
+        return value.researchResource
+      })[0]
+  }
+
+  getOrganizationDisambiguated(host: Host): OrgDisambiguated {
+    const disambiguationSource = host.disambiguationSource
+    const orgDisambiguatedId = host.orgDisambiguatedId
+
+    return this.detailsOrgDisambiguated
+      .filter(
+        (value) =>
+          value.disambiguationSource === disambiguationSource &&
+          value.orgDisambiguatedId === orgDisambiguatedId
+      )
+      .map((value) => {
+        return value.orgDisambiguated
+      })[0]
+  }
+
+  isPreferred(
+    researchResource: ResearchResource,
+    group: ResearchResourcesGroup
+  ) {
+    const response =
+      researchResource && group.defaultResearchResource[0]
+        ? group[0].defaultResearchResource.putCode.value ===
+          researchResource.putCode.value
+        : false
+    return response
+  }
+
+  getLink(type: string, value: string) {
+    return this._organizationsService.getLink(type, value)
```

Comment:
Seems like this is important, but never use? 
If not require we can remove the function from the service as well

Reply:
https://github.com/ORCID/orcid-angular/pull/736/files#r610050341